src/core/Provider.tsx

```diff
@@ -58,8 +58,9 @@ export default class Provider extends React.Component<Props, State> {
       const theme =
         this.state.colorScheme === 'light' ? DefaultTheme : DarkTheme;
 
-      return Object.assign(theme as ReactNativePaper.Theme, {
+      return Object.assign({}, theme as ReactNativePaper.Theme, {
         animation: {
+          ...theme.animation,
           scale: reduceMotionEnabled ? 0 : 1,
         },
       });
```

Comment:
Instead of mixing `Object.assign` and spread operator, it's better to use one:

```suggestion
      return {
        ...theme,
        animation: {
          ...theme.animation,
          scale: reduceMotionEnabled ? 0 : 1,
        },
      };
```